classify-cluster-priority/README.md

```diff
@@ -0,0 +1,85 @@
+- As Giant Swarm operations staff I want to understand the importance of clusters so I can make appropriate decisions, e. g. when prioritizing problem mitigation in an incident affecting multiple clusters.
+
+## Current practice
+
+- One customer uses the first character of the cluster name to distinguish between production (`p`), development (`d`), and staging `s`. This customer also has clusters which don't fall into either of these categories and their names start wtith arbotrary letters.
+- Many customers use specific terms and phrases in the cluster description, e. g. "Production", "test", "testing", "ignore alerts", "prd", "prod", "nonprod".
+- Some customers use labels on the `clusters.cluster.x-k8s.io` resource (the cluster's main resource) indicating the importance. E. g.
+    - `odp/environment: development`
+    - `odp/environment: staging`
+    - `odp/environment: production`.
+
+## Proposal
+
+We introduce a label and a schema for the label values to indicate the relative importance of a cluster.
+
+Label name: `giantswarm.io/service-priority`
+
+The label is to be used on the main resource defining the cluster, usually of resource type `clusters.cluster.x-k8s.io`.
+
+To simplify maintenance and avoid logical conflicts, this label SHOULD NOT be applied on other resources of the cluster. Systems MUST ignore the label used on other resources in the same cluster. This means that it is not possible, for example, to differentiate between the importance of sevewral node pools within the same cluster.
+
+### Well-defined label values
+
+There are three well-defined label values to be used.
+
+- `giantswarm.io/service-priority: production`: A cluster labelled this way serves production traffic (i. e. end-user facing). Clusters with this label value are considered the most important ones, relative to other clusters not carrying this label.
+
+- `giantswarm.io/service-priority: staging`: Clusters carrying this label are used to simulate running on a production cluster, but are not serving production traffic. They are considered less important than ones carrying the above `production` label.
+
+- `giantswarm.io/service-priority: development`: A cluster with this label is used by developers to develop and test applications. It is considered less important than a cluster labelled as `production` or `staging`.
+
+### Absence of the label
+
+The absence of the label SHOULD be interpreted as if the label was present with the value `production`.
+
+### Using undefined values
+
+If the label is used with a value different from the well-defined ones, the label SHOULD be interpreted as if the value was `development`.
+
+## Additional considerations and explanations
+
+- Understanding of this system by Giant Swarm staff is crucial. We have to make sure that everybody in operations know about the system.
+
+- Individual customers may or may not use the system. It is not required that all customers adopt it for it to become usable.
```

Comment:
If customers don't use this, do the clusters get the default labels?
If so, I would think labels should default to low, so as not to overwhelm the people doing operations and to encourage customers to use the label scheme.

Reply:
We currently treat every cluster equally important (unless we actually have some inside knowledge of what it's used for) so defaulting to critical wouldn't change anything.

Reply:
Makes sense, thanks for the clarification.